From 0d77b2d21cea4c12dd8534f2df6d7e1358e12367 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Tue, 8 Sep 2020 12:47:49 +0800
Subject: [PATCH] dt-bindings: sound: es8311: add some optional properties

This patch add adc-volume/dac-volume/aec-mode and be easy to
configure some simple functions without alsa-utils.

Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
Change-Id: Ic58188a0ef72fe5236b523445dc530e649a0e248
---
 .../devicetree/bindings/sound/es8311.txt         | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/sound/es8311.txt b/Documentation/devicetree/bindings/sound/es8311.txt
index 0c1afb1b56b8..e93747817488 100644
--- a/Documentation/devicetree/bindings/sound/es8311.txt
+++ b/Documentation/devicetree/bindings/sound/es8311.txt
@@ -9,7 +9,18 @@ Optional properties:
 
 - clocks: The phandle of the master clock to the CODEC
 - clock-names: Should be "mclk"
-
+- adc-volume: The volume of ADC, range is: 0x00(-95dB) ~ 0xff(+32dB), 0dB is 0xbf.
+- dac-volume: The volume of DAC, range is: 0x00(-95dB) ~ 0xff(+32dB), 0dB is 0xbf.
+- aec-mode: The string of description AEC path between ADC and DAC, It should be:
+		"adc left, adc right",
+		"adc left, null right",
+		"null left, adc right",
+		"null left, null right",
+		"dac left, adc right",
+		"adc left, dac right",
+		"dac left, dac right"
+	    And aec-mode is "adc left, adc right" by default, if the property
+	    is not specified.
 Example:
 
 es8311: es8311@18 {
@@ -17,5 +28,8 @@ es8311: es8311@18 {
 	reg = <0x18>;
 	clocks = <&cru MCLK_I2S0_TX_OUT2IO>;
 	clock-names = "mclk";
+	adc-volume = <0xbf>;  /* 0dB */
+	dac-volume = <0xbf>;  /* 0dB */
+	aec-mode = "dac left, adc right";
 	spk-ctl-gpios = <&gpio2 5 GPIO_ACTIVE_HIGH>;
 };
-- 
2.35.3

